src/mono/mono/utils/atomic.h

```diff
@@ -25,7 +25,231 @@ F/MonoDroid( 1568): shared runtime initialization error: Cannot load library: re
 Apple targets have historically being problematic, xcode 4.6 would miscompile the intrinsic.
 */
 
-#if defined(HOST_WIN32)
+#if defined(_MSC_VER)
+
+#define MONO_IGNORE_STDATOMIC 1
+#endif
+
+#if defined(HOST_ANDROID) && (defined(HOST_X86) || defined(HOST_AMD64) || defined(HOST_ARM))
```

Comment:
I think the opt-in setup would make more sense here

Reply:
I'm not so sure. I would expect the number of platforms that don't support stdatomic to decrease over time.

I can try it both ways.  (And I want to move this decision to cmake to make it more visible)

Reply:
Off by default seems ok